backend/dataall/modules/redshift_datasets_shares/services/redshift_table_share_processor.py

```diff
@@ -349,14 +356,12 @@ def process_revoked_shares(self) -> bool:
                     share_item = ShareObjectRepository.find_sharable_item(
                         self.session, self.share.shareUri, table.rsTableUri
                     )
-                    if not self.reapply:
-                        failed_table_SM = ShareItemSM(started_state)
-                        failed_state = failed_table_SM.run_transition(ShareItemActions.Failure.value)
-                        failed_table_SM.update_state_single_item(self.session, share_item, failed_state)
-                    else:
-                        ShareStatusRepository.update_share_item_health_status(
-                            self.session, share_item, ShareItemHealthStatus.Unhealthy.value, str(e), datetime.now()
-                        )
+                    failed_table_SM = ShareItemSM(started_state)
+                    failed_state = failed_table_SM.run_transition(ShareItemActions.Failure.value)
+                    failed_table_SM.update_state_single_item(self.session, share_item, failed_state)
+                    ShareStatusRepository.update_share_item_health_status(
+                        self.session, share_item, ShareItemHealthStatus.Unhealthy.value, str(e), datetime.now()
+                    )
```

Comment:
should we set to UnHealthy if revoke fails? We do not do so for other revokes but I think makes sense

Reply:
I changed it while implementing :) If we were able to rollback all the steps before failure then we could leave it as healthy, but given that in the revoke we will break the share we can change it to unhealthy. I'll do